feat(otelecho): add SpanNameFormatter config and tests (#6365)
open-telemetry/opentelemetry-go-contrib: CHANGELOG.md

- Added support for providing `endpoint`, `pollingIntervalMs` and `initialSamplingRate` using environment variable `OTEL_TRACES_SAMPLER_ARG` in `go.opentelemetry.io/contrib/samples/jaegerremote`. (#6310)
- Added support exporting logs via OTLP over gRPC in `go.opentelemetry.io/contrib/config`. (#6340)

### Changed

> - Change the logic of span to make it comply with OTEL specifications in `go.opentelemetry.io/contrib/instrumentation/github.com/labstack/echo/otelecho`. (#6365)

Reviewer: No need to specify it's the otel spec. We're within otel, that's assumed :)
Also, someone reading this may not know what the spec says. So mentioning the new value is good.

```suggestion
- Change the span name to be `GET /path` so it complies with the specifications in `go.opentelemetry.io/contrib/instrumentation/github.com/labstack/echo/otelecho`. (#6365)
```

Reviewer 2: _I edited @dmathieu suggestion so that it references semantic conventions instead of the specification._
